lib/Dialect/Openfhe/Transforms/ConfigureCryptoContext.cpp

```diff
@@ -56,9 +56,23 @@ SmallVector<int64_t> findAllRotIndices(func::FuncOp op) {
   return rotIndicesResult;
 }
 
+// Helper function to check if the function has BootstrapOp
+bool hasBootstrapOp(func::FuncOp op) {
+  bool result = false;
+  op.walk<WalkOrder::PreOrder>([&](Operation *op) {
+    if (isa<openfhe::BootstrapOp>(op)) {
+      result = true;
+      return WalkResult::interrupt();
+    }
+    return WalkResult::advance();
+  });
+  return result;
+}
```

Comment:
I think I will clean these up after this PR goes in, since we have a few implementations of "walk until you find fn(op)" in the codebase.